contracts/colony/ColonyFunding.sol

```diff
@@ -421,10 +421,10 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs { // ignore-swc-123
     require(block.timestamp - payout.blockTimestamp <= 60 days, "colony-reward-payout-not-active");
 
     ITokenLocking tokenLocking = ITokenLocking(IColonyNetwork(colonyNetworkAddress).getTokenLocking());
-    uint256 userDepositTimestamp = tokenLocking.getUserLock(token, msg.sender).timestamp;
+    uint256 userDepositLockCount = tokenLocking.getUserLock(token, msg.sender).lockCount;
     uint256 userTokens = tokenLocking.getUserLock(token, msg.sender).balance;
 
-    require(userDepositTimestamp < payout.blockTimestamp, "colony-reward-payout-deposit-too-recent");
+    require(userDepositLockCount < payoutId, "colony-reward-payout-lock-count-too-high");
```

Comment:
We really mean that `userDepositLockCount + 1 == payoutId` right? It's meant to be exactly one less, not just arbitrarily less.

Reply:
Yeah, we do, I _think_ this can just be straight up removed though as the appropriate checks are done in `unlockTokenForUser`. I think that realisation also solves the problem I've been worrying about existing colonies that would check the timestamp here with these changes, and find zero (because new deposits don't get timestamps / timestamps aren't updated with the new TokenLocking), but unlockToken covers us there.

I still don't know why we do all this in `calculateRewardForUser` though :sweat_smile:

Reply:
Agreed, that check already exists in TokenLocking... although I could see it clarifying for an error to be here as well (otherwise it seems sort of hidden). And yup, we've definitely discussed how `calculateRewardForUser` should be a `view` function and all the state changes should be elsewhere, but /shrug.

Reply:
We get `colony-token-locking-has-previous-active-locks` instead of `colony-token-locking-has-previous-active-locks` which seems pretty much the same to me.

Reply:
You mean instead of `colony-reward-payout-lock-count-too-high`?

Reply:
I do, yes, got a bit trigger happy with my pasting there.